data/tutorials/language/0lg_08_env.md

```diff
@@ -0,0 +1,557 @@
+- A function taking two strings, returning their concatenation, with a space character in between
+- A function taking a list, returning `None` if the list is empty, and the tail of the list otherwise
+
+### Defining Global Functions
+
+Defining a global function isn't very different from creating a definition and binding a value to a name. The only difference is that no computation takes place. The expression is turned into a function value and bound to the name.
+```ocaml
+# let f = fun x -> x * x;;
+val f : int -> int = <fun>
+
+# let f x = x * x;;
+val f : int -> int = <fun>
+```
+
+These two definitions are the same. The former explicitly binds the anonymous function to a name. The latter is the same, but it uses a more compact syntax and avoids the `fun`` keyword and the arrow symbol.
+
+### Defining Local Functions
+
+A function may be defined locally. Just like any local definition, the function is only available inside its attached expression. Although the local functions are often defined inside the scope of the function, this is not a requirement. Here is a local function inside a toplevel expression.
+```ocaml
+# let sq x = x * x in sq 7 * sq 7;;
+- : int = 2401
+```
+
+### Closures
+
+The following sequence of definitions and expressions illustrates what a closure is.
+```ocaml
+# let d = 2 * 3;;
+val d : int = 6
+
+# let f x = x * d;;
+val f : 'a -> int = <fun>
+
+# f 7;;
+- : int = 42
+
+# let d = 10;;
+val d : int = 10
+
+# f 7;; (* What the result? *)
+- : int = 42
+```
+
+Here is how this makes sense:
+1. Constant `d` is defined, and its value is 6.
+1. Function `f` is defined. It takes a single parameter and returns its product by `d`
+1. Compute `f` of 7
+1. Create a new definition `d` shadowing the first one
+1. Compute `f` of 7 again, the result is the same
+
+Although the new definition of `d` shadows the first one, the first value of `d` remains the one used inside `f`. That first definition of `d` is trapped inside the definition of `f`. Inside `f` the environment is frozen as it was when `f` was defined, with the first value of `d` inside. That's why it is called a closure.
+
+Partially applying parameters also creates a closure.
+```ocaml
+# let max_42 = max 42;;
+val max_42 : int -> int = <fun>
+```
+
+In the `max_42` function, the environment contains an additional binding between the first parameter of `max` and the value 42.
+
+Actually, all function values are closures.
+
+### Recursive Functions
+
+In order to perform iterated computations, a function may call itself. Such a function is called _recursive_. Recursive functions must be defined; they can't be anonymous. A recursive function also needs to be explicitly declared as such by using `let rec`. In OCaml, either a function is declared as recursive or it isn't recursive. It is not possible to accidentally create recursion loops between functions. As a consequence, recursive functions can't be anonymous.
```

Comment:
```suggestion
In order to perform iterated computations, a function may call itself. Such a function is called _recursive_. Recursive functions must be defined; therefore, they can't be anonymous. A recursive function also needs to be explicitly declared as such by using `let rec`, so it is not possible to accidentally create recursion loops between functions. In OCaml, unless a function is declared recursive, it isn't recursive by default.
```

Would it be more clear to say this? It removes the repetition at the end, but I'm not sure this is still accurate.